src/GPU3D.cpp

```diff
@@ -569,7 +533,17 @@ void GPU3D::SetEnabled(bool geometry, bool rendering) noexcept
     if (!rendering) ResetRenderingState();
 }
 
-
+inline void GPU3D::BeginWriteFD(u16 cmd, u32* param)
+{
+    if (NDS.GPU.FD != nullptr)
+    {
+        if (NDS.GPU.FD->FDWrite(cmd, param))
+        {
+            NDS.GPU.FD = nullptr;
+            // add an osd notif here?
```

Comment:
I think the frontend would be a better place for an OSD notification. The frontend is calling this method in the first place, right?